### ISSUE TITLE
Broken  Text classification demo small_imdb_reviews

### ISSUE THREAD

**Issue by disdi** [czhu12/light_bulb#1]
$ python code/server.py --config config/small_imdb_reviews.yml
Using TensorFlow backend.
2018-09-04 11:23:27.350124: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
WARNING:tensorflow:From /home/saket/anaconda2/envs/py36/lib/python3.6/site-packages/keras/backend/tensorflow_backend.py:1213: calling reduce_max (from tensorflow.python.ops.math_ops) with keep_dims is deprecated and will be removed in a future version.
Instructions for updating:
keep_dims is deprecated, use keepdims instead
WARNING:tensorflow:From /home/saket/anaconda2/envs/py36/lib/python3.6/site-packages/keras/backend/tensorflow_backend.py:2885: calling reduce_sum (from tensorflow.python.ops.math_ops) with keep_dims is deprecated and will be removed in a future version.
Instructions for updating:
keep_dims is deprecated, use keepdims instead
WARNING:tensorflow:From /home/saket/anaconda2/envs/py36/lib/python3.6/site-packages/keras/backend/tensorflow_backend.py:1349: calling reduce_mean (from tensorflow.python.ops.math_ops) with keep_dims is deprecated and will be removed in a future version.
Instructions for updating:
keep_dims is deprecated, use keepdims instead
Layer (type)                 Output Shape              Param #   
embedding_1 (Embedding)      (None, None, 50)          20000150  
dropout_1 (Dropout)          (None, None, 50)          0         
conv1d_1 (Conv1D)            (None, None, 250)         37750     
activation_1 (Activation)    (None, None, 250)         0         
dropout_2 (Dropout)          (None, None, 250)         0         
conv1d_2 (Conv1D)            (None, None, 250)         187750    
global_max_pooling1d_1 (Glob (None, 250)               0         
dense_1 (Dense)              (None, 250)               62750     
dropout_3 (Dropout)          (None, 250)               0         
activation_2 (Activation)    (None, 250)               0         
dense_2 (Dense)              (None, 2)                 502       
activation_3 (Activation)    (None, 2)                 0         
Total params: 20,288,902
Trainable params: 288,752
Non-trainable params: 20,000,150
Started local server at http://localhost:5000
2018-09-04 11:23:27,761 - label_app - DEBUG - Need at least 40 labels to start labelling
 * Serving Flask app "server" (lazy loading)
 * Environment: production
   WARNING: Do not use the development server in a production environment.
   Use a production WSGI server instead.
 * Debug mode: on
2018-09-04 11:23:27,767 - label_app - DEBUG - {'labelled': {'total': 0, 'train': 0, 'model_labelled': 0, 'test': 0}, 'unlabelled': 2000}
2018-09-04 11:23:27,775 - label_app - ERROR - Not ready to train, only has 0 samples, needs 40 samples.
2018-09-04 11:23:27,775 - label_app - ERROR - Backing off interval time...
2018-09-04 11:23:27,785 - label_app - ERROR - Not ready to test, only has 0 samples, needs 40 samples.
2018-09-04 11:23:27,786 - label_app - DEBUG - Starting representation training.
2018-09-04 11:23:27,790 - label_app - DEBUG - Finished representation training. Loss: None
 * Running on http://127.0.0.1:5000/ (Press CTRL+C to quit)
2018-09-04 11:23:37,775 - label_app - DEBUG - Need at least 40 labels to start labelling
2018-09-04 11:23:42,815 - label_app - DEBUG - {'labelled': {'total': 0, 'train': 0, 'model_labelled': 0, 'test': 0}, 'unlabelled': 2000}
2018-09-04 11:23:42,820 - label_app - ERROR - Not ready to train, only has 0 samples, needs 40 samples.
2018-09-04 11:23:42,820 - label_app - ERROR - Backing off interval time...
2018-09-04 11:23:42,825 - label_app - ERROR - Not ready to test, only has 0 samples, needs 40 samples.
2018-09-04 11:23:42,826 - label_app - DEBUG - Starting representation training.
2018-09-04 11:23:42,828 - label_app - DEBUG - Finished representation training. Loss: None
2018-09-04 11:23:57,798 - label_app - DEBUG - Need at least 40 labels to start labelling
2018-09-04 11:24:05,359 - label_app - DEBUG - {'labelled': {'total': 0, 'train': 0, 'model_labelled': 0, 'test': 0}, 'unlabelled': 2000}
2018-09-04 11:24:05,364 - label_app - ERROR - Not ready to train, only has 0 samples, needs 40 samples.
2018-09-04 11:24:05,364 - label_app - ERROR - Backing off interval time...
2018-09-04 11:24:05,367 - label_app - ERROR - Not ready to test, only has 0 samples, needs 40 samples.
2018-09-04 11:24:05,368 - label_app - DEBUG - Starting representation training.
2018-09-04 11:24:05,370 - label_app - DEBUG - Finished representation training. Loss: None
2018-09-04 11:24:27,832 - label_app - DEBUG - Need at least 40 labels to start labelling
2018-09-04 11:24:35,407 - label_app - DEBUG - {'labelled': {'total': 0, 'train': 0, 'model_labelled': 0, 'test': 0}, 'unlabelled': 2000}
2018-09-04 11:24:35,413 - label_app - ERROR - Not ready to train, only has 0 samples, needs 40 samples.
2018-09-04 11:24:35,413 - label_app - ERROR - Backing off interval time...
2018-09-04 11:24:35,416 - label_app - ERROR - Not ready to test, only has 0 samples, needs 40 samples.
2018-09-04 11:24:35,416 - label_app - DEBUG - Starting representation training.
2018-09-04 11:24:35,418 - label_app - DEBUG - Finished representation training. Loss: None

**Comment by czhu12** [comment]
This is expected, can you go to localhost:5000? The model needs 40 labels before it can start training
